keps/sig-network/4858-ip-cidr-validation/README.md

```diff
@@ -0,0 +1,838 @@
+      UNRESOLVED, because the "compare the values as strings" thing
+      doesn't work if you have to worry about some objects having
+      legacy non-canonical values.)
+
+<<[/UNRESOLVED]>>
+
+
+<<[UNRESOLVED LoadBalancerSourceRanges ]>>
+
+- MAYBE additionally tighten LoadBalancerSourceRanges validation to
+  remove the historical accident that values are allowed to be
+  whitespace padded. Because you could argue that the current
+  situation with that field could also potentially lead to security
+  issues if one agent is able to parse the LoadBalancerSourceRanges
+  value but another fails (because it wasn't expecting whitespace) and
+  so decides the value is unset.
+
+<<[/UNRESOLVED]>>
+
+
+<<[UNRESOLVED non-ip-fields ]>>
+
+- MAYBE tighten/make-more-consistent the validation of other
+  networking-related fields at the same time. In particular, there are
+  some fields that are supposed to be validated as being "hostnames"
+  but which accidentally end up being validated as "either hostname or
+  IPv4 address" (because they use `utilvalidation.IsDNS1123Subdomain`,
+  which requires basically "alphanumeric separated by dots", which
+  IPv4 addresses are). This could be fixed with a
+  `utilvalidation.IsHostname` that allows hostnames but rejects IP
+  addresses (but this would also be a potentially-breaking change that
+  would require ratcheting).
+
+<<[/UNRESOLVED]>>
+
+
+<<[UNRESOLVED non-special-ip ]>>
+
+- MAYBE revisit the use of `ValidateNonSpecialIP`: certain kinds of
+  IPs (loopback, multicast, link-local, etc) do not make sense in many
+  contexts, but we only prohibit them in a few places (endpoints and
+  external IPs).
+
+<<[/UNRESOLVED]>>
+
+
+<<[UNRESOLVED cli-validation ]>>
+
+- MAYBE do something about the validation of IP/CIDR values in CLI
+  args / component configs, since these also sometimes get passed to
+  external APIs. There is less room for changing the mandatory
+  validation here without possibly completely breaking people on
+  upgrade, but we could at least (a) try to ensure that *new* IP/CIDR
+  config fields/args get validated correctly, and (b) consistently
+  warn when old IP/CIDR config fields/args have invalid values. (e.g.,
+  we could have `utilvalidation.IsValidLegacyIP` that accepts invalid
+  IPs but logs or returns a warning message.)
+
+<<[/UNRESOLVED]>>
+```
+
+[already done for IPs and CIDRs in Service]: https://github.com/kubernetes/kubernetes/blob/v1.31.1/pkg/api/service/warnings.go
+[CEL IP/CIDR validation helpers]: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/cel/library/ip.go
+
+### Non-Goals
+
+- Updating/replacing `netutils.ParseIPSloppy` and
+  `netutils.ParseCIDRSloppy`. (There is a separate plan for this.)
+
+- Providing `netip.Addr`/`netip.Prefix`-based utilities or increasing
+  the usage of those types. (There is a separate plan for this.)
+
+- (Any of the UNRESOLVEDs above that we decide not to do.)
```

Comment:
What is the planned for unresolved items? When they will be resolved/decided?

Reply:
I was hoping @thockin would express some opinions before we merged but he's been busy...

I guess all of the UNRESOLVED Goals are initially assumed to not be Goals.